### PR TITLE
Fix: pkg/core/adapters can not be used with pkg/next

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin <denis.tingajkin@xored.com>

### Motivation
https://github.com/networkservicemesh/sdk/issues/61

### CODE REVIEW

**Comment by edwarnicke** [comment]
Try rebasing... now that things are using networkservicemesh/api not networkservicemesh/networkservicemesh/controlplane/api you will need it :)

**Comment by denis-tingajkin** [comment]
@edwarnicke thanks, the problems are fixed.

File: pkg/networkservice/core/next/client.go [hunk 1]
```diff
@@ -50,6 +52,9 @@ func NewWrappedNetworkServiceClient(wrapper ClientWrapper, clients ...networkser
 
 // NewNetworkServiceClient - chains together clients into a single networkservice.NetworkServiceClient
 func NewNetworkServiceClient(clients ...networkservice.NetworkServiceClient) networkservice.NetworkServiceClient {
+	if len(clients) == 0 {
```
Comment (edwarnicke) [inline pkg/networkservice/core/next/client.go]: Nice catch :)

File: pkg/networkservice/core/next/context.go [hunk 2]
```diff
@@ -47,6 +51,9 @@ func Server(ctx context.Context) networkservice.NetworkServiceServer {
 	if rv, ok := ctx.Value(nextServerKey).(networkservice.NetworkServiceServer); ok {
 		return rv
 	}
+	if rv, ok := ctx.Value(nextClientKey).(networkservice.NetworkServiceClient); ok {
```
Comment (edwarnicke) [inline pkg/networkservice/core/next/context.go]: So I'm curious, why are we converting nextClient to nextServer here?  Perhaps I still need my morning coffee... but I've lost the thread :)

Comment (denis-tingajkin) [reply pkg/networkservice/core/next/context.go]: It is needed for the case when we have for example a chain of servers and one adpated client
```
s := next.NewNetworkServiceServer(
something.NewServer(), 
adapters.NewClientToServer(something.NewClient()),
something.NewServer())
```
And we want that this adapted client can correctly call the next server.

Note: This client in `Request()` and `Close()`  uses  `next.Client(ctx)`. So if we will not adapt here we will break a chain.

Comment (edwarnicke) [reply pkg/networkservice/core/next/context.go]: Excellent!  Thank you for the patient explanation :)

**Pull request merged**